Source/SwiftLintFramework/Rules/NimbleOperatorRule.swift

```diff
@@ -39,25 +40,118 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule {
-        let excludingKinds = SyntaxKind.commentKinds()
+    typealias Operators = (to: String?, toNot: String?)
+    typealias MatcherFunction = String
 
-        let matches = file.matchPattern(pattern).filter { _, kinds in
-            // excluding comment kinds and making sure first token (`expect`) is an identifier
-            kinds.filter(excludingKinds.contains).isEmpty && kinds.first == .identifier
-        }
+    fileprivate let operatorsMapping: [MatcherFunction: Operators] = [
+        "equal": (to: "==", toNot: "!="),
+        "beIdenticalTo": (to: "===", toNot: "!=="),
+        "beGreaterThan": (to: ">", toNot: nil),
+        "beGreaterThanOrEqualTo": (to: ">=", toNot: nil),
+        "beLessThan": (to: "<", toNot: nil),
+        "beLessThanOrEqualTo": (to: "<=", toNot: nil)
+    ]
 
+    public func validateFile(_ file: File) -> [StyleViolation] {
+        let matches = violationMatchesRanges(in: file)
         return matches.map {
             StyleViolation(ruleDescription: type(of: self).description,
                 severity: configuration.severity,
-                location: Location(file: file, characterOffset: $0.0.location))
+                location: Location(file: file, characterOffset: $0.location))
         }
     }
+
+    private func violationMatchesRanges(in file: File) -> [NSRange] {
+        let operatorNames = operatorsMapping.keys
+        let operatorsPattern = "(" + operatorNames.joined(separator: "|") + ")"
+
+        let variablePattern = "(.(?!expect\\())+?"
+        let pattern = "expect\\(\(variablePattern)\\)\\.to(Not)?\\(\(operatorsPattern)\\(\(variablePattern)\\)\\)"
+
+        let excludingKinds = SyntaxKind.commentKinds()
+
+        return file.matchPattern(pattern)
+            .filter { _, kinds in
+                kinds.filter(excludingKinds.contains).isEmpty && kinds.first == .identifier
+            }.map { $0.0 }
+    }
+
+    public func correctFile(_ file: File) -> [Correction] {
+        let matches = violationMatchesRanges(in: file)
+            .filter { file.ruleEnabledViolatingRanges([$0], forRule: self).isEmpty == false }
+        guard matches.isEmpty == false else { return [] }
+
+        let description = type(of: self).description
+        var corrections: [Correction] = []
+        var contents = file.contents
+
+        matchesLoop: for range in matches.sorted(by: { $0.location > $1.location }) {
+            for (functionName, operatorCorrections) in operatorsMapping {
+                guard let correctedString = contents.replace(function: functionName,
+                                                             with: operatorCorrections,
+                                                             in: range)
+                else {
+                    continue
+                }
+
+                contents = correctedString
+                corrections.append(Correction(ruleDescription: description,
```

Comment:
also, since you're iterating over the matches in last-to-first order (i.e. in reverse), to have the corrections in first-to-last order, you'd need to `corrections.insert(..., at: 0)` instead.